internal/storage/storage_handle.go

```diff
@@ -261,35 +280,32 @@ func NewStorageHandle(ctx context.Context, clientConfig storageutil.StorageClien
 		return nil, fmt.Errorf("could not create StorageControl Client: %w", err)
 	}
 
-	// ShouldRetry function checks if an operation should be retried based on the
-	// response of operation (error.Code).
-	// RetryAlways causes all operations to be checked for retries using
-	// ShouldRetry function.
-	// Without RetryAlways, only those operations are checked for retries which
-	// are idempotent.
-	// https://github.com/googleapis/google-cloud-go/blob/main/storage/storage.go#L1953
-	retryOpts := []storage.RetryOption{storage.WithBackoff(gax.Backoff{
-		Max:        clientConfig.MaxRetrySleep,
-		Multiplier: clientConfig.RetryMultiplier,
-	}),
-		storage.WithPolicy(storage.RetryAlways),
-		storage.WithErrorFunc(storageutil.ShouldRetry)}
-
-	hc.SetRetry(retryOpts...)
-	gc.SetRetry(retryOpts...)
-
-	// The default MaxRetryAttempts value is 0 indicates no limit.
-	if clientConfig.MaxRetryAttempts != 0 {
-		hc.SetRetry(storage.WithMaxAttempts(clientConfig.MaxRetryAttempts))
-		gc.SetRetry(storage.WithMaxAttempts(clientConfig.MaxRetryAttempts))
-	}
-
 	sh = &storageClient{
-		httpClient:           hc,
-		grpcClient:           gc,
 		storageControlClient: controlClient,
-		clientProtocol:       clientConfig.ClientProtocol,
-		directPathDetector:   directPathDetector,
+		clientConfig:         clientConfig,
+		directPathDetector:   &gRPCDirectPathDetector{clientOptions: clientOpts},
+	}
+	return
+}
+
+func (sh *storageClient) getClient(ctx context.Context, isbucketZonal bool) (sc *storage.Client, err error) {
+	if isbucketZonal {
+		if sh.grpcClientWithBidiConfig == nil {
+			sh.grpcClientWithBidiConfig, err = createGRPCClientHandle(ctx, &sh.clientConfig, true)
```

Comment:
instead of some elseif cases, can you simply it a bit. improves readability and easy to extend in future.

if isBucketZonal {
if sh.grpcClientWithBidiConfig == nil {
			sh.grpcClientWithBidiConfig, err = createGRPCClientHandle(ctx, &sh.clientConfig, true)
return sh.grpcClientWithBidiConfig, nil
}

if sh.clientConfig.ClientProtocol == cfg.GRPC {
		if sh.grpcClient == nil {
			sh.grpcClient, err = createGRPCClientHandle(ctx, &sh.clientConfig, false)
		}
	return sh.grpcClient, nil
	}